File: frontend/src/types/quotation.ts
export type QuotationStatus = 'draft' | 'finalized' | 'converted' | 'cancelled';

export interface QuotationItem {
  id: string;
  productId: string;
  productSku: string;
  productName: string;
  quantity: number;
  unitPrice: number;
  discountPct: number;
  lineTotal: number;
}

export interface Quotation {
  id: string;
  quotationNumber: string;
  customerId?: string;
  customerName?: string;
  salespersonId: string;
  salespersonName: string;
  status: QuotationStatus;
  subtotal: number;
  discountPct: number;
  discountAmount: number;
  totalAmount: number;
  notes?: string;
  validUntil?: string;
  finalizedAt?: string;
  cancelledAt?: string;
  cancellationReason?: string;
  convertedToSaleId?: string;
  items: QuotationItem[];
  createdAt: string;
}

export interface CreateQuotationRequest {
  customerId?: string;
  notes?: string;
  validUntil?: string;
}

export interface AddQuotationItemRequest {
  productId: string;
  quantity: number;
  discountPct?: number;
}

export interface FinalizeQuotationRequest {
  discountPct?: number;
  notes?: string;
}

export interface CancelQuotationRequest {
  cancellationReason: string;
}

File: backend/Stoctable.Domain/Entities/Quotation.cs
using Stoctable.Domain.Entities.Base;
using Stoctable.Domain.Enums;

namespace Stoctable.Domain.Entities;

public class Quotation : BaseEntity
{
    public string QuotationNumber { get; set; } = string.Empty;
    public Guid? CustomerId { get; set; }
    public Guid? SalespersonId { get; set; }
    public QuotationStatus Status { get; set; } = QuotationStatus.Draft;
    public decimal Subtotal { get; set; }
    public decimal DiscountPct { get; set; }
    public decimal DiscountAmount { get; set; }
    public decimal TotalAmount { get; set; }
    public string? Notes { get; set; }
    public DateOnly? ValidUntil { get; set; }
    public DateTimeOffset? FinalizedAt { get; set; }
    public DateTimeOffset? CancelledAt { get; set; }
    public string? CancellationReason { get; set; }
    public Guid? ConvertedToSaleId { get; set; }

    public Customer? Customer { get; set; }
    public User? Salesperson { get; set; }
    public ICollection<QuotationItem> Items { get; set; } = [];
    public ICollection<StockReservation> StockReservations { get; set; } = [];
}
